### ISSUE TITLE
[Bug] Unsupported version of dbt-core installed using pip

### ISSUE THREAD

**Issue by dhodge250** [oracle/dbt-oracle#116]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Not sure if this needs to be a separate issue from #108, but the [requirements](https://github.com/oracle/dbt-oracle/blob/main/setup.py#L42) section in the `setup.py` file set's the dbt-core version to whatever is greater than `v1.6`
`requirements = [
        "dbt-core~=1.6",
        "cx_Oracle==8.3.0",
        "oracledb==1.4.1"
]`. We recently upgraded to `dbt-oracle v1.6.0` on 10/29/2023 right after `dbt-core v1.7.0` was released, and pip upgraded dbt-core to the latest version (`v1.7.0`) rather than the dbt-oracle currently supported version of `v1.6.7`. This dbt-core upgrade broke our dbt environments until we could downgrade back to `v1.6.7`. This issue would break other's dbt environments if they were to upgrade dbt-oracle right now.

### Expected Behavior

When upgrading from any version prior to `dbt-oracle v1.6.0` to the latest version, I would expect pip to upgrade the `dbt-core` package to the latest version supported by `dbt-oracle`, not to a version that isn't supported by `dbt-oracle`.

### Steps To Reproduce

1. Setup a clean environment of `dbt-oracle` with any version prior to `v1.6.0`
2. Upgrade `dbt-oracle` to the latest version using this command: `python -m pip install dbt-oracle --upgrade`
3. Confirm upgraded package of `dbt-core`

### Environment

```markdown
- OS:Windows 10
- Python: 3.9.11
- dbt: dbt-core 1.6.7
       dbt-oracle 1.6.0
```

### What Oracle database version are you using dbt with?

19c

**Comment by aosingh** [comment]
@dhodge250 this is fixed in `dbt-oracle==1.6.1`

**Comment by dhodge250** [comment]
@aosingh thanks for jumping on this so quickly!
